hudi-common/src/main/java/org/apache/hudi/common/util/ParquetUtils.java

```diff
@@ -149,13 +150,26 @@ public static BloomFilter readBloomFilterFromParquetMetadata(Configuration confi
         readParquetFooter(configuration, false, parquetFilePath,
             HoodieAvroWriteSupport.HOODIE_AVRO_BLOOM_FILTER_METADATA_KEY,
             HoodieAvroWriteSupport.OLD_HOODIE_AVRO_BLOOM_FILTER_METADATA_KEY,
-            HoodieAvroWriteSupport.HOODIE_BLOOM_FILTER_TYPE_CODE);
+            HoodieAvroWriteSupport.HOODIE_BLOOM_FILTER_TYPE_CODE,
+            HoodieAvroWriteSupport.HOODIE_BLOOM_FILTER_IS_COMPRESSED,
+            HoodieAvroWriteSupport.HOODIE_BLOOM_FILTER_COMPRESSION_TYPE);
     String footerVal = footerVals.get(HoodieAvroWriteSupport.HOODIE_AVRO_BLOOM_FILTER_METADATA_KEY);
     if (null == footerVal) {
       // We use old style key "com.uber.hoodie.bloomfilter"
       footerVal = footerVals.get(HoodieAvroWriteSupport.OLD_HOODIE_AVRO_BLOOM_FILTER_METADATA_KEY);
     }
     BloomFilter toReturn = null;
+    boolean isCompressed = false;
+    if (footerVals.containsKey(HoodieAvroWriteSupport.HOODIE_BLOOM_FILTER_IS_COMPRESSED)) {
+      isCompressed = Boolean.valueOf(footerVals.get(HoodieAvroWriteSupport.HOODIE_BLOOM_FILTER_IS_COMPRESSED));
+      if (isCompressed) {
+        String compressionType = footerVals.get(HoodieAvroWriteSupport.HOODIE_BLOOM_FILTER_COMPRESSION_TYPE);
+        Preconditions.checkArgument(compressionType.equals(GzipCompressionUtils.TYPE),
```

Comment:
this can be replaced with ValidationUtils.checkArgument() once the PR# 1159 has been merged

Reply:
Sounds good.